rotkehlchen/chain/evm/decoding/merkl/utils.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from typing import Any
+
+from rotkehlchen.errors.misc import RemoteError, UnableToDecryptRemoteData
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChainID
+from rotkehlchen.utils.network import request_get
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _paginated_request(url_template: str, items_per_page: int, context: str, **url_params: Any) -> list | None:  # noqa: E501
+    """Make paginated requests to Merkl API.
+
+    `url_template` is the URL template with {page} and {items} placeholders
+    `items_per_page` is the number of items per page
+    `context` is the context string for error logging
+    `**url_params` are the additional parameters for URL formatting
+
+    Returns a list of all items from all pages, or None if there was an error.
+    """
+    all_items: list[Any] = []
+    page = 0
+    while True:
+        try:
+            url = url_template.format(page=page, items=items_per_page, **url_params)
+            all_items.extend(response := request_get(url))
+            if len(response) == 0 or len(response) < items_per_page:  # If we get less than the full page size, we've reached the end  # noqa: E501
+                break
+
+            page += 1
+
+        except (RemoteError, UnableToDecryptRemoteData, KeyError) as e:
+            msg = f'missing key {e!s}' if isinstance(e, KeyError) else str(e)
+            log.error(f'Failed to retrieve {context} page {page} due to {msg}')
+            return None
+
+    log.debug(f'Retrieved {len(all_items)} items for {context} across {page + 1} pages')
+    return all_items
+
+
+def get_merkl_protocol_for_token(
+        account: str,
+        token: str,
+        chain_id: ChainID,
+) -> str | None:
+    """Get protocol for a specific account and token by fetching directly from Merkl API"""
+    if (opportunities := _paginated_request(
+        url_template='https://api.merkl.xyz/v4/opportunities?chainId={chain_id}&search={token}&items={items}&status=LIVE,PAST,SOON',
+        items_per_page=100,
+        context=f'Merkl opportunities for account {account=} on {chain_id=}',
+        account=account,
+        chain_id=chain_id.serialize_for_db(),
+        token=token,
+    )) is None:
+        return None
+
+    protocols, seen = [], set()
+    for opportunity in opportunities:
+        try:
+            if (protocol := opportunity['protocol']['name']) in seen:
+                continue
+
+            protocols.append(protocol)
+            seen.add(protocol)
+        except KeyError as e:
+            log.error(f'Failed to get protocols for {account=} on {chain_id=} due to missing key {e!s}')  # noqa: E501
+            continue
+
+    # if we're not sure where the claim is from, we return None.
```

Comment:
i opened a ticket in the merkl discord as to why what's returned from the api is different from what the app shows.

once they respond, it removes the uncertainty and we can just make a new api request using the `campaignId`.